main.tf

```diff
@@ -3,13 +3,14 @@ terraform {
 }
 
 locals {
-  is_domain_name    = length(regexall("[.]", var.bucket_name)) > 0 # contains a dot/period ?
-  public_read       = local.is_domain_name ? true : var.public_read
-  uniform_access    = local.is_domain_name ? true : var.uniform_access
-  enable_versioning = local.is_domain_name ? true : var.enable_versioning
-  bucket_name       = local.is_domain_name ? var.bucket_name : format("%s-%s", var.bucket_name, var.name_suffix)
-  bucket_labels     = merge(var.labels, { "name_suffix" = var.name_suffix })
-  bucket_location   = var.location != "" ? var.location : data.google_client_config.google_client.region
+  is_domain_name        = length(regexall("[.]", var.bucket_name)) > 0 # contains a dot/period ?
+  public_read           = local.is_domain_name ? true : var.public_read
+  uniform_access        = local.is_domain_name ? true : var.uniform_access
+  enable_versioning     = local.is_domain_name ? true : var.enable_versioning
+  bucket_name           = local.is_domain_name ? var.bucket_name : format("%s-%s", var.bucket_name, var.name_suffix)
+  bucket_labels         = merge(var.labels, { "name_suffix" = var.name_suffix })
+  bucket_location       = var.location != "" ? var.location : data.google_client_config.google_client.region
+  sanitized_domain_name = format("%s-%s", replace(var.bucket_name, ".", "-"), var.name_suffix)
```

Comment:
IMO `sanitized_domain_name` should **_just_** be the sanitized version of **_just_** the domain name - not more - not less.

We may opt to add the `name_suffix` into the "format" of each of the resource names themselves.

What do you think?

Reply:
we can do that, but then for all resources need to do `name = format("%s-%s", sanitized_domain_name, var.name_suffix)` which feels redundant

Reply:
Since we are using it as a suffix everywhere, we may declare a new local called `resource_name_suffix` which will be concatenation of the `local.sanitized_domain_name` + `var.name_suffix` and then use that everywhere to avoid repitition.

My main concern is, `sanitized_domain_name` should _just_ be the sanitized version of _just_ the domain name - not more - not less.

Reply:
updated